contracts/Colony.sol

```diff
@@ -74,12 +74,54 @@ contract Colony is ColonyStorage, PatriciaTreeProofs {
     setRoleAssignmentFunction(bytes4(keccak256("setTaskEvaluatorRole(uint256,address)")));
     setRoleAssignmentFunction(bytes4(keccak256("setTaskWorkerRole(uint256,address)")));
 
+    // Setting this so there will be no tokens available for issuing when the colony is created
+    totalIssuedInSeconds = now;
+
     // Initialise the root domain
     IColonyNetwork colonyNetwork = IColonyNetwork(colonyNetworkAddress);
     uint256 rootLocalSkill = colonyNetwork.getSkillCount();
     initialiseDomain(rootLocalSkill);
   }
 
+  function setTokenIssuanceRate(uint256 _amount, uint256 _period, uint256 _precision) public auth {
+    require(_amount > 0 && _period > 0, "token-issuance-rate-bad-inputs");
+    require(sub(now, lastIssuenceRateChangeTimestamp) >= 4 weeks, "token-issuance-rate-change-not-allowed");
+
+    // If we are setting the rate for the first time we will not go through these checks
+    if (tokenIssuanceRate.amount > 0 && tokenIssuanceRate.period > 0) {
+      // Calculate the current rate in form of token/second
+      uint256 oldRate = mul(tokenIssuanceRate.amount, _precision) / tokenIssuanceRate.period;
+      // Calculate new rate
+      uint256 newRate = mul(_amount, _precision) / _period;
+      // this will throw if `_precision` is not high enough
+      require(oldRate > 0 && newRate > 0, "token-issuance-rate-bad-precision");
+      // Get the difference between rates
+      uint256 diff = (oldRate > newRate) ? sub(oldRate, newRate) : sub(newRate, oldRate);
+      require(diff > 0, "token-issuance-rate-invalid-rate");
+      // How much is new rate different (in percent) relative to the current one
+      uint256 percent = mul(diff, 100) / oldRate;
+      // Must be less than 10 percent
+      require(percent <= 10, "token-issuance-rate-change-too-big");
+    }
+
+    lastIssuenceRateChangeTimestamp = now;
+
```

Comment:
Additionally here call `mintTokens` for the full remaining amount allowed to be minted as otherwise this would cause the new issuance rate to be "backdated" to the last mint time under the previous rate.

Reply:
Done.

But what do you think about just setting `totalAmountIssuedUntilTimestamp = now`. This way, user will be able to choose whether they want to mint remaining tokens before they change the rate.

Reply:
Good point. Sorry I saw this just now, maybe I prefer your idea to what I suggested above. Let's switch to that.

Reply:
Just add a note in natspec for describing that we restart the time for token issuance rate when this is set.